File: CubeRootFinder.java
public class CubeRootFinder {
    public static void main(String[] args) {
        int target = 42;
        findCubeRoots(target);
    }

    public static void findCubeRoots(int target) {
        int cubeRoot = (int) Math.round(Math.pow(target, 1.0 / 3));

        for (int i = -cubeRoot; i <= cubeRoot; i++) {
            for (int j = -cubeRoot; j <= cubeRoot; j++) {
                for (int k = -cubeRoot; k <= cubeRoot; k++) {
                    if (i * i * i + j * j * j + k * k * k == target) {
                        System.out.printf("The three numbers whose cube is %d are: %d, %d, and %d%n", target, i, j, k);
                        return;
                    }
                }
            }
        }

        System.out.printf("No three numbers found whose cube is %d%n", target);
    }
}

File: cube_root_finder.py
import math

def find_cube_roots(target):
    cube_root = round(target ** (1/3))

    for i in range(-cube_root, cube_root + 1):
        for j in range(-cube_root, cube_root + 1):
            for k in range(-cube_root, cube_root + 1):
                if i ** 3 + j ** 3 + k ** 3 == target:
                    print(f"The three numbers whose cube is {target} are: {i}, {j}, and {k}")
                    return

    print(f"No three numbers found whose cube is {target}")

# Example usage
find_cube_roots(42)
